### DISCUSSION TITLE
Allow to choose between Procfile.dev or Puma plugin

### DISCUSSION THREAD

**Discussion by javierav** [rails/tailwindcss-rails#327] [Ideas]
after the PR #300 has been merged, the user should be allowed to choose whether to use the `Procfile.dev` or the Puma plugin in the installation generator. Currently the `Procfile.dev` and the `foreman`gem installation are always used.

**Comment by flavorjones** [comment]
@javierav This is a great idea! If you'd like to attempt to implement it, I'd gladly review a PR.

**Reply by javierav** [reply]
Hi Mike!

I started thinking about it, but I don't see what is the best way to parameterize the generator. Any suggestions?

**Comment by javierav** [comment]
#331

**Comment by flavorjones** [comment]
I think, long-term, the real solution here is to use the engine server hook approach from #347. I'll be working on landing some version of that patch.
